### PR TITLE
For #235: Implementing create with additional parameters

### PR DESCRIPTION
**Pull request by bkuzmic**
For #235:

- finished implementing create with additional parameters

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#239](https://github.com/amihaiemil/docker-java-api/issues/239) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#239](https://github.com/amihaiemil/docker-java-api/issues/239) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/0242b449-23e9-4ae6-9cd8-fc7aa7bf2a8e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/main/java/com/amihaiemil/docker/RtNetworks.java [hunk 1]
```diff
@@ -27,9 +27,7 @@
 
 import java.io.IOException;
 import java.net.URI;
-import javax.json.Json;
-import javax.json.JsonObject;
-import javax.json.JsonObjectBuilder;
+import javax.json.*;
```
Comment (paulodamaso) [inline src/main/java/com/amihaiemil/docker/RtNetworks.java]: @bkuzmic It's not a good practice to use `*` imports

File: src/main/java/com/amihaiemil/docker/RtNetworks.java [hunk 2]
```diff
@@ -74,50 +72,16 @@
...
-            create.releaseConnection();
-        }
+        return this.create(name, Json.createObjectBuilder().build());
     }
 
     @Override
     public Network create(final String name, final JsonObject filters)
         throws IOException, UnexpectedResponseException {
-        throw new UnsupportedOperationException("create not implemented");
+        JsonObjectBuilder json = Json.createObjectBuilder();
```
Comment (paulodamaso) [inline src/main/java/com/amihaiemil/docker/RtNetworks.java]: @bkuzmic Let's make this guy `final`

File: src/main/java/com/amihaiemil/docker/RtNetworks.java [hunk 3]
```diff
@@ -155,4 +119,50 @@ HttpClient client() {
...
+            new HttpPost(
+                String.format("%s/%s", this.baseUri.toString(), "create")
+            );
+        try {
+            create.setEntity(
+                new StringEntity(
+                    json.build().toString(), ContentType.APPLICATION_JSON
+                )
+            );
+            JsonObject createResult = this.client.execute(
```
Comment (paulodamaso) [inline src/main/java/com/amihaiemil/docker/RtNetworks.java]: @bkuzmic Let's make this guy final too

File: src/main/java/com/amihaiemil/docker/RtNetworks.java [hunk 4]
```diff
@@ -155,4 +119,50 @@ HttpClient client() {
...
+                create,
+                new ReadJsonObject(
+                    new MatchStatus(
+                        create.getURI(),
+                        HttpStatus.SC_CREATED
+                    )
+                )
+            );
+            if (createResult.size() > 0) {
```
Comment (paulodamaso) [inline src/main/java/com/amihaiemil/docker/RtNetworks.java]: @bkuzmic Let's use `isEmpty` method

File: src/test/java/com/amihaiemil/docker/RtNetworksTestCase.java [hunk 5]
```diff
@@ -152,71 +150,57 @@ public void createWithParametersOk() throws Exception {
...
-                ),
-                new Condition(
-                    "create() must send a POST HTTP request",
-                    req -> "POST".equals(req.getRequestLine().getMethod())
-                ),
-                new Condition(
-                    "create() must send the request to the create url",
-                    req -> "http://localhost/networks/create".equals(
-                        req.getRequestLine().getUri()
+        Network network =
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtNetworksTestCase.java]: @bkuzmic `final`...

File: src/test/java/com/amihaiemil/docker/RtNetworksTestCase.java [hunk 6]
```diff
@@ -152,71 +150,57 @@ public void createWithParametersOk() throws Exception {
...
+                    "Labels",
+                    labels
+                ).build()
+            );
         MatcherAssert.assertThat(
             "could not return correct network id",
             network.getString("Id"),
             new IsEqual<>("id1")
         );
-        MatcherAssert.assertThat(
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtNetworksTestCase.java]: @bkuzmic I think that these assertions here are easier to understand than using that `Condition` construct

Comment (bkuzmic) [reply src/test/java/com/amihaiemil/docker/RtNetworksTestCase.java]: @paulodamaso I agree, but here I'm testing the request payload and in the response I only get Id back.

**Comment by paulodamaso** [comment]
@bkuzmic Great, thanks. About that unresolved comment, I now see that we are testing our request payload sent, so it can stay this way.

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Code review was too long (12 days), architects (@amihaiemil) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#239](https://github.com/amihaiemil/docker-java-api/issues/239) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Job `gh:amihaiemil/docker-java-api#239` doesn't exist in WBS, can't create order

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**